python/lsst/phosim/utils/phosim_repackager.py

```diff
@@ -97,110 +101,111 @@ def repackage(self, phosim_amp_files, out_dir='.'):
         phosim_amp_files: list
             List of phosim amplifier filenames.
         """
-        # PhoSim labels the amplifier channels incorrectly.  Here is a
-        # mapping from the phosim channels to the correct ones.
-        ch_map = {'00': '10',
-                  '01': '11',
-                  '02': '12',
-                  '03': '13',
-                  '04': '14',
-                  '05': '15',
-                  '06': '16',
-                  '07': '17',
-                  '17': '07',
-                  '16': '06',
-                  '15': '05',
-                  '14': '04',
-                  '13': '03',
-                  '12': '02',
-                  '11': '01',
-                  '10': '00'}
-
-        # Create the HDUList to contain the MEF data.
-        sensor = fits.HDUList(fits.PrimaryHDU())
 
-        # Extract the data for each segment from the FITS files
-        # into a dictionary keyed by channel id.
-        segments = dict()
-        for fn in phosim_amp_files:
-            # Get the channel id from the filename written by phosim.
-            # This is the incorrect channel id, so correct it when
-            # filling the segments dictionary.
-            phosim_channel = os.path.basename(fn).split('_')[6][1:]
-            segments[ch_map[phosim_channel]] = fits.open(fn)[0]
+        segments = {}
+        for filename in phosim_amp_files:
+            phosim_channel = os.path.basename(filename).split('_')[6][1:]
+            segments[phosim_channel] = fits.open(filename)[0]
+
+        # Read raft name from the first filename
+        filename = phosim_amp_files[0]
+        raft = os.path.basename(filename).split('_')[4]
+        ccdslot = os.path.basename(filename).split('_')[5]
+        sensorId = '%s_%s'%(raft, ccdslot)
+        print('Repackaging sensor %s '%(sensorId))
+
+        detectors = self.camera.get(sensorId)
+        TELCODE = self.telcode
+        CONTRLLR = self.contrllr
```

Comment:
Why do you need these two? Redundancy? If these two attributes are fixed, you could define them as the class attributes with the capital names instead.

Reply:
I wanted it to be clearer by saying  eg `sensor[0].header['TELCODE'] = TELCODE`. If you'd rather it said `sensor[0].header['TELCODE'] = self.TELCODE` I can change that

Reply:
Changed the class attributes to capital names (`self.telcode` to `self.TELCODE` ) , and instead of assigning in each function `TELCODE = self.TELCODE`, I use `self.TELCODE` directly.

Reply:
I think this should work. But another way is:
```
class FooClass:
    TELCODE = "foo"
    
    def __init__(self):
        self.attr = "bar"
```

This is what I mean for the class attribute:
https://www.toptal.com/python/python-class-attributes-an-overly-thorough-guide

Reply:
Ok, in this case I'm setting the class attribute `TELCODE = "foo" ` because it is a constant.